posthog/api/__init__.py

```diff
@@ -92,6 +84,7 @@ def api_not_found(request):
 router.register(r"login", authentication.LoginViewSet)
 router.register(r"reset", authentication.PasswordResetViewSet, "password_reset")
 router.register(r"users", user.UserViewSet)
+router.register(r"staff_users", user.UserStaffViewSet)
```

Comment:
Couldn't this feature be operated just via the existing `/api/users`?

Reply:
That was what I did originally but we don't support patching users on `/api/users` other than yourself, and adding that functionality there would've required a bunch of extra validations to ensure no weird behavior. We could still use this and only use the endpoint to update `is_staff`, in the end they're staff users with maximum permissions.

Reply:
A long (for PostHog) time ago I moved us from `/api/user` to `/api/users` in part in case we need some user management solution like this, so from the API experience perspective I strongly recommend using `UserViewSet` for this. Basically something like this https://github.com/PostHog/posthog/pull/8192, just raising a permissions error if an update payload contains `is_staff` and the requesting user is not `is_staff`.

Reply:
Yup, pretty much what I had before, will update, thx @Twixes!